src/main/java/org/folio/entlinks/service/InstanceLinkService.java

```diff
@@ -41,6 +45,29 @@ public void updateInstanceLinks(UUID instanceId, @NotNull InstanceLinkDtoCollect
     repository.saveAll(linksToCreate);
   }
 
+  @Transactional
+  public LinkCountMapDtoCollection countNumberOfTitles(UuidCollection authorityIdCollection) {
+    var ids = authorityIdCollection.getIds();
+    var linkCountMap = repository.countNumberOfTitlesByAuthorityIds(ids)
+        .stream().map(mapper::convert).toList();
+
+    linkCountMap = fillInMissingIdsWithZeros(linkCountMap, ids);
+
+    return new LinkCountMapDtoCollection().links(linkCountMap);
+  }
+
+  private List<LinkCountMapDto> fillInMissingIdsWithZeros(List<LinkCountMapDto> linksCountMap, List<UUID> ids) {
+    var foundIds = linksCountMap.stream().map(LinkCountMapDto::getId).toList();
+    var notFoundIds = ids.stream().filter(uuid -> foundIds.stream().noneMatch(uuid::equals)).toList();
+
+    if (!notFoundIds.isEmpty()) {
+      var tempList = new ArrayList<>(linksCountMap);
+      notFoundIds.forEach(uuid -> tempList.add(new LinkCountMapDto().id(uuid).totalLinks(0L)));
+      linksCountMap = tempList;
+    }
```

Comment:
What is the intent of creating a temp list?

Reply:
because linksCountMap is immutable, we should use different reference